crates/ruff/src/rules/flake8_todo/rules.rs

```diff
@@ -0,0 +1,384 @@
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): fix some issue
+/// ```
+#[violation]
+pub struct MissingTextInTodo;
+impl Violation for MissingTextInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing text in TODO")
+    }
+}
+
+/// ## What it does
+/// Checks that a "TODO" tag is properly capitalized, i.e. that the tag is uppercase.
+///
+/// ## Why is this bad?
+/// Inconsistent capitalization leads to less readable code.
+///
+/// ## Example
+/// ```python
+/// # todo(ruff): capitalize this
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): this is capitalized
+/// ```
+#[violation]
+pub struct InvalidCapitalizationInTodo {
+    pub tag: String,
+}
+impl AlwaysAutofixableViolation for InvalidCapitalizationInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let InvalidCapitalizationInTodo { tag } = self;
+        format!("Invalid TODO capitalization: `{tag}` should be `TODO`")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Fix capitalization in `TODO`".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks that the colon after a "TODO" tag is followed by a space.
+///
+/// ## Why is this bad?
+/// Skipping the space after a colon leads to less readable code.
+///
+/// ## Example
+/// ```python
+/// # TODO(ruff):fix this
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): fix this
+/// ```
+#[violation]
+pub struct MissingSpaceAfterColonInTodo;
+impl Violation for MissingSpaceAfterColonInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing space after colon in TODO")
+    }
+}
+
+// Capture groups correspond to checking:
+//      1. Tag (used to match against `TODO` in capitalization and spelling, e.g. ToDo and FIXME)
+//      2. Author exists (in parentheses) with 0 or 1 spaces between it and the tag, but before colon
+//      3. Colon exists after author
+//      4. Space exists after colon
+//      5. Text exists after space
+//
+// We can check if any of these exist in one regex. Capture groups that don't pick anything up
+// evaluate to `None` in Rust, so the capture group index will always correspond to its respective rule
+// whether the token has been found or not.
+//
+// Example:
+// ```python
+// # TODO(evanrittenhouse): This is a completely valid TODO
+// ```
+// will yield [Some("TODO") Some("evanrittenhouse"), Some(":"), Some(" "), Some("This is a completely valid TODO")], whereas
+// ```python
+// # ToDo this is completely wrong
+// ```
+// will yield [Some("ToDo"), None, None, Some(" "), Some("this is completely wrong")]. Note the
+// `Nones` for the colon and space checks.
+//
+// Note: Regexes taken from https://github.com/orsinium-labs/flake8-todos/blob/master/flake8_todos/_rules.py#L12.
+static TODO_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^# {0,1}(?P<tag>[tT][oO][dD][oO]|BUG|FIXME|XXX)( {0,1}\(.*\))?(:)?( )?(.+)?$")
```

Comment:
Can we use a case-insensitive match instead of writing out all the different combinations of TODO? Should we also match on `fixme` or `FixMe`?